client/src/components/Product-items/Products.js

```diff
@@ -0,0 +1,118 @@
+import React from "react";
+import { connect } from "react-redux";
+import { gql } from "apollo-boost";
+import { Query, Mutation } from "react-apollo";
+
+import { onAddedItem } from "../../redux/shopCart/cart-actions";
+
+import {
+  ProductContainer,
+  ProductCard,
+  MediaImage,
+  ProductFooterContainer,
+  ProductInfoContainer,
+  AddButton,
+  LoadingProduct
+} from "./products-styles.js";
+
+const LIKE_PRODUCT = gql`
+  mutation LikeProduct($userId: String!, $productId: String!) {
+    likeProduct(userId: $userId, productId: $productId) {
+      userId
+    }
+  }
+`;
+
+const GET_PRODUCTS = gql`
+  {
+    merchants {
+      guid
+      merchant
+      products {
+        id
+        name
+        price
+        description
+        color
+        size
+        image
+        quantity
+      }
+    }
+  }
+`;
+
+const withLike = Component => props => {
+  return (
+    <Mutation mutation={LIKE_PRODUCT}>
+      {like => (
+        <Component
+          onLike={(userId, productId) =>
+            like({ variables: { userId, productId } })
+          }
+          {...props}
+        />
+      )}
+    </Mutation>
+  );
+};
+
+const withProducts = Component => props => {
+  return (
+    <Query query={GET_PRODUCTS}>
+      {({ loading, data }) => {
+        return (
+          <Component
+            merchantsLoading={loading}
+            merchants={data && data.merchants}
+            {...props}
+          />
+        );
+      }}
+    </Query>
+  );
+};
+
+const ProductsList = ({ merchants, merchantsLoading, onAddedItem, onLike }) => {
```

Comment:
refactored to functional components from the original class components that was provided